File: src/main/java/net/softbell/bsh/domain/converter/AuthStatusRuleConverter.java
package net.softbell.bsh.domain.converter;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

import net.softbell.bsh.domain.AuthStatusRule;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 인증 상태 자료형 DB 컨버터
 */
@Converter(autoApply = true)
public class AuthStatusRuleConverter implements AttributeConverter<AuthStatusRule, Integer>
{
	@Override
	public Integer convertToDatabaseColumn(AuthStatusRule authStatusRule)
	{
		if (authStatusRule == null)
			return null;
		
		return authStatusRule.getCode();
	}

	@Override
	public AuthStatusRule convertToEntityAttribute(Integer code)
	{
		return AuthStatusRule.ofLegacyCode(code);
	}
}

File: src/main/kotlin/net/softbell/bsh/domain/converter/AuthStatusRuleConverter.kt
package net.softbell.bsh.domain.converter

import net.softbell.bsh.domain.AuthStatusRule
import javax.persistence.AttributeConverter
import javax.persistence.Converter

/**
 * @author : Bell(bell@softbell.net)
 * @description : 인증 상태 자료형 DB 컨버터
 */
@Converter(autoApply = true)
class AuthStatusRuleConverter : AttributeConverter<AuthStatusRule, Int> {
    override fun convertToDatabaseColumn(authStatusRule: AuthStatusRule): Int {
        return authStatusRule.code
    }

    override fun convertToEntityAttribute(code: Int): AuthStatusRule? {
        return AuthStatusRule.ofLegacyCode(code)
    }
}